lib/Cryptol.cry

```diff
@@ -323,3 +324,87 @@ groupBy : {each,parts,elem} (fin each) =>
   [parts * each] elem -> [parts][each]elem
 groupBy = split`{parts=parts}
 
+/**
+ * Logical implication
+ */
+(==>) : {a} a -> a -> a
+a ==> b = not a || b
+
+/**
+ * Logical negation
+ */
+not : {a} a -> a
+not a = ~ a
+
+/**
+ * Conjunction
+ */
+and : {n} (fin n) => [n] -> Bit
+and xs = ~zero == xs
+
+/**
+ * Disjuntion
+ */
+or : {n} (fin n) => [n] -> Bit
+or xs = zero != xs
+
```

Comment:
Hmm, on one hand, `not` is more consistent with `and` and `or` with the more polymorphic type. But it is also just a prefix synonym for `~`. Is your thought that `not` will just signal to the reader  that we're thinking logically rather than bitwise?